src/stores/courses.js

```diff
@@ -1,40 +1,36 @@
-import { reactive } from 'vue';
 import { defineStore } from 'pinia';
-import courseData from '../assets/data/courses.json';
+import { reactive } from 'vue';
 
-export const useCoursesStore = defineStore('courses', () => {
-	let courses = reactive(courseData[0].courses);
+export const useCoursesStore = (courseData, courseId) => {
```

Comment:
The function in `defineStore` is sadly something we can't modify, so rather than pass args like this:

```
export const useCoursesStore = defineStore('courses', (courseData, courseId) => {
```

We have to instead wrap `defineStore` with another function. This creates the closure that we want though. Now, callsites only ever have to pass in `courseId` once. We also pass in `courseData` so that this can be easily tested.

Reply:
I will add that coupling a store to an API like this is probably not great (doubly so since the API is "fixed" now to a single course). It's arguably better to separate the two, such that you're defining a store (presumably this is just a centralized place for data?) and then an API to operate on that store. But I don't know Vue/Pinia/etc., so maybe there's a good reason for this?

Regardless, this is just an example PR to demonstrate how to simplify the design a bit. The concept can easily be translated to a more "production-ready" context.